src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -9,8 +11,19 @@
 public class UndoCommand extends Command {
     public static final String COMMAND_WORD = "undo";
 
+    private ElisaStateHistory elisaStateHistory;
+
+    public UndoCommand(ElisaStateHistory elisaStateHistory) {
+        this.elisaStateHistory = elisaStateHistory;
+    }
+
     @Override
     public CommandResult execute(ItemModel model) throws CommandException {
-        return new CommandResult("Hello from undo");
+        if (elisaStateHistory.size() <= 1) {
+            throw new CommandException(Messages.MESSAGE_NOTHING_TO_UNDO);
+        } else {
+            elisaStateHistory.popCommand();
+            return new CommandResult("Undo successful!", true);
```

Comment:
Just as a add on, maybe it would be good to store the previous command as well so that the user will know what command they undo. However, that can be an extension, and not needed for this pull request.

Reply:
I will be refactoring to remove the need for time-travelling, so I will implement this in a subsequent PR :)